### DISCUSSION TITLE
Permissions needed to List All API Tokens for All Users

### DISCUSSION THREAD

**Discussion by jcresp21** [netbox-community/netbox#13359] [Q&A]
Hi everyone!

I have using and developing an app that integrates with Netbox ( our current running version is v3.3.8).
We create API Tokens on the go to use certain Netbox API functions, we set an expiration date of 1 hour after the API token creation, that help us force the user to log in again with the app.

I have created a Service Account which purpose should be **only** deleting expired API tokens ( a python script using pynetbox),  The problem is that  I can't seem to find the correct permission to list all API Tokens for All Users) , I don't think there is such permission btw, the only thing that seemed to work was setting the user account as SuperUser, but obviously that's not the desired solution, as it allows that service account to use any api endpoint without restrictions.

Has anyone come across with a problem similar to this ? 

Thanks in advance!

**Comment by abhi1693** [comment]
Have you tried assigning `view` on `Users > Token`?

**Reply by jcresp21** [reply]
Hi abhi, yes, I have in fact tried assigging all permissions but it seems like this functionality is only allowed to "superuser"  ( marked in the user settings) , wondering if this could be a good request, but now that I think of it, it would be difficult to show tokens to a user that has less permissions . Maybe only showing the id and being able to delete the tokens ?

**Reply by abhi1693** [reply]
At least on v3.6 beta1, this works if the permission is assigned
![image](https://github.com/netbox-community/netbox/assets/5083532/22b90be4-dcf7-490b-8548-1288fd91b4fb)
![image](https://github.com/netbox-community/netbox/assets/5083532/e0bc1482-a79c-4c7e-99a0-f58ce20363bb)
![image](https://github.com/netbox-community/netbox/assets/5083532/5438ab75-99c9-4ea9-91c8-b1c8cd1e2c6d)
![image](https://github.com/netbox-community/netbox/assets/5083532/ec680898-9330-4d2b-a279-003134019ed2)
